diff --git a/utils/qemu/Makefile b/utils/qemu/Makefile
index bee745052..9d20c9ab1 100644
--- a/utils/qemu/Makefile
+++ b/utils/qemu/Makefile
@@ -28,8 +28,8 @@ PKG_BUILD_DEPENDS+=spice-protocol
 include $(INCLUDE_DIR)/nls.mk
 include $(INCLUDE_DIR)/package.mk
 
-QEMU_DEPS_IN_GUEST := @(TARGET_x86_64||TARGET_armsr||TARGET_malta)
-QEMU_DEPS_IN_HOST := @(TARGET_x86_64||TARGET_sunxi)
+QEMU_DEPS_IN_GUEST := @(TARGET_x86_64||TARGET_armsr||TARGET_malta||aarch64)
+QEMU_DEPS_IN_HOST := @(TARGET_x86_64||TARGET_sunxi||aarch64)
 QEMU_DEPS_IN_HOST += +libstdcpp
 QEMU_DEPS_IN_HOST += $(ICONV_DEPENDS)
 
@@ -220,6 +220,7 @@ endef
 qemu-target-list :=	\
 	x86_64-softmmu	\
 	arm-softmmu	\
+	aarch64-softmmu	\
 
 qemu-target-x86_64-softmmu-deps:= \
 	+qemu-firmware-pxe \
