From 40676fd44153c3d153db0fef3075daf665990764 Mon Sep 17 00:00:00 2001
From: sbwml <admin@cooluc.com>
Date: Wed, 17 Jul 2024 17:46:42 +0800
Subject: [PATCH] backends: fix string initialization error on gcc15

Signed-off-by: sbwml <admin@cooluc.com>
---
 backends/i386_regs.c   | 2 +-
 backends/x86_64_regs.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/backends/i386_regs.c b/backends/i386_regs.c
index 7ec93bb..4bca1b1 100644
--- a/backends/i386_regs.c
+++ b/backends/i386_regs.c
@@ -83,7 +83,7 @@ i386_register_info (Ebl *ebl __attribute__ ((unused)),
 
   switch (regno)
     {
-      static const char baseregs[][2] =
+      static const char baseregs[][3] =
 	{
 	  "ax", "cx", "dx", "bx", "sp", "bp", "si", "di", "ip"
 	};
diff --git a/backends/x86_64_regs.c b/backends/x86_64_regs.c
index ef987da..c92c862 100644
--- a/backends/x86_64_regs.c
+++ b/backends/x86_64_regs.c
@@ -80,7 +80,7 @@ x86_64_register_info (Ebl *ebl __attribute__ ((unused)),
 
   switch (regno)
     {
-      static const char baseregs[][2] =
+      static const char baseregs[][3] =
 	{
 	  "ax", "dx", "cx", "bx", "si", "di", "bp", "sp"
 	};
-- 
2.43.5

