diff --git a/package/utils/util-linux/Makefile b/package/utils/util-linux/Makefile
index b385030e39..82ccd02d5e 100644
--- a/package/utils/util-linux/Makefile
+++ b/package/utils/util-linux/Makefile
@@ -511,6 +511,15 @@ define Package/wipefs/description
  libblkid.
 endef
 
+define Package/taskset
+$(call Package/util-linux/Default)
+  TITLE:=Set or retrieve a process's CPU affinity
+endef
+
+define Package/taskset/description
+  Alter the priority of running processes
+endef
+
 CONFIGURE_ARGS += \
 	--disable-use-tty-group		\
 	--disable-rpath			\
@@ -791,6 +800,11 @@ define Package/wipefs/install
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/wipefs $(1)/usr/sbin/
 endef
 
+define Package/taskset/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/taskset $(1)/usr/bin/
+endef
+
 # these lines need to be ordered by dependency because of ABI versioning
 $(eval $(call BuildPackage,libuuid))
 $(eval $(call BuildPackage,libblkid))
@@ -835,3 +849,4 @@ $(eval $(call BuildPackage,uuidgen))
 $(eval $(call BuildPackage,wall))
 $(eval $(call BuildPackage,whereis))
 $(eval $(call BuildPackage,wipefs))
+$(eval $(call BuildPackage,taskset))
