--- a/feeds/luci/modules/luci-base/htdocs/luci-static/resources/ui.js
+++ b/feeds/luci/modules/luci-base/htdocs/luci-static/resources/ui.js
@@ -4142,7 +4142,7 @@ var UI = baseclass.extend(/** @lends LuCI.ui.prototype */ {
 							'class': 'btn',
 							'click': function() {
 								UI.prototype.hideModal();
-								rejectFn(new Error('Upload has been cancelled'));
+								rejectFn(new Error(_('Upload has been cancelled')));
 							}
 						}, [ _('Cancel') ]),
 						' ',
