--- a/src/apk_adb.c
+++ b/src/apk_adb.c
@@ -177,7 +177,6 @@ static struct adb_scalar_schema scalar_n
 
 static adb_val_t version_fromstring(struct adb *db, apk_blob_t val)
 {
-	if (!apk_version_validate(val)) return ADB_ERROR(APKE_PKGVERSION_FORMAT);
 	return adb_w_blob(db, val);
 }
 
@@ -338,12 +337,6 @@ static int dependency_fromstring(struct
 	apk_blob_t bname, bver;
 	int op;
 
-	if (apk_dep_parse(bdep, &bname, &op, &bver) != 0) goto fail;
-	if ((op & APK_DEPMASK_CHECKSUM) != APK_DEPMASK_CHECKSUM &&
-	    !apk_version_validate(bver)) goto fail;
-
-	if (apk_blob_spn(bname, APK_CTYPE_DEPENDENCY_NAME, NULL, NULL)) goto fail;
-
 	adb_wo_blob(obj, ADBI_DEP_NAME, bname);
 	if (op != APK_DEPMASK_ANY) {
 		adb_wo_blob(obj, ADBI_DEP_VERSION, bver);
