diff --git a/util.c b/util.c
index 9eba483..98be8b3 100644
--- a/util.c
+++ b/util.c
@@ -21,6 +21,7 @@
 
 #include <crypto/scatterwalk.h>
 #include <linux/scatterlist.h>
+#include <linux/version.h>
 #include "util.h"
 
 /* These were taken from Maxim Levitsky's patch to lkml.
@@ -45,7 +46,11 @@ struct scatterlist *sg_advance(struct scatterlist *sg, int consumed)
 
 	if (sg->offset >= PAGE_SIZE) {
 		struct page *page =
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 18, 0)
+			page = sg_page(sg) + (sg->offset / PAGE_SIZE);
+#else
 			nth_page(sg_page(sg), sg->offset / PAGE_SIZE);
+#endif
 		sg_set_page(sg, page, sg->length, sg->offset % PAGE_SIZE);
 	}
 
