diff --git a/utils/qemu/Makefile b/utils/qemu/Makefile
index a706993d0..e1454782b 100644
--- a/utils/qemu/Makefile
+++ b/utils/qemu/Makefile
@@ -29,8 +29,8 @@ include $(INCLUDE_DIR)/uclibc++.mk
 include $(INCLUDE_DIR)/nls.mk
 include $(INCLUDE_DIR)/package.mk
 
-QEMU_DEPS_IN_GUEST := @(TARGET_x86_64||TARGET_armvirt||TARGET_malta)
-QEMU_DEPS_IN_HOST := @(TARGET_x86_64||TARGET_sunxi)
+QEMU_DEPS_IN_GUEST := @(TARGET_x86_64||TARGET_armvirt||TARGET_malta||aarch64)
+QEMU_DEPS_IN_HOST := @(TARGET_x86_64||TARGET_sunxi||aarch64)
 QEMU_DEPS_IN_HOST += $(CXX_DEPENDS)
 QEMU_DEPS_IN_HOST += $(ICONV_DEPENDS)
 
@@ -218,6 +218,7 @@ endef
 qemu-target-list :=	\
 	x86_64-softmmu	\
 	arm-softmmu	\
+	aarch64-softmmu	\
 
 qemu-target-x86_64-softmmu-deps:= \
 	+qemu-firmware-pxe \
