From 04d0a5456c87c6ce7a77bef18b26e4b6dbb05265 Mon Sep 17 00:00:00 2001
From: Benny Baumann <BenBE@geshi.org>
Date: Mon, 9 Dec 2024 22:18:08 +0100
Subject: [PATCH] Avoid compilation issues with ncurses on GCC 15

Fixes: #1567
---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

--- a/configure.ac
+++ b/configure.ac
@@ -449,6 +449,7 @@ if test "$my_htop_platform" = "solaris";
 fi
 AC_CHECK_FUNCS( [set_escdelay] )
 AC_CHECK_FUNCS( [getmouse] )
+AC_DEFINE([NCURSES_ENABLE_STDBOOL_H], [1], [Define to enable stdbool.h in ncurses])
 
 
 AC_ARG_ENABLE([affinity],
