From 0bfe29ac8cefc3335184e93e5ae0a03bdaaff336 Mon Sep 17 00:00:00 2001
From: sbwml <admin@cooluc.com>
Date: Sun, 30 Jun 2024 21:58:24 +0800
Subject: [PATCH 04/11] rootfs: add r/w permissions for UCI configuration files

Signed-off-by: sbwml <admin@cooluc.com>
---
 include/rootfs.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/rootfs.mk b/include/rootfs.mk
index eb69a99..f8bc2ed 100644
--- a/include/rootfs.mk
+++ b/include/rootfs.mk
@@ -100,6 +100,7 @@ define prepare_rootfs
 			$(STAGING_DIR_HOST)/bin/upx --lzma --best "$(1)$$file" || true; \
 		done < "$(TOPDIR)/upx_list.txt"; \
 	fi
+	chmod 600 $(1)/etc/config/*
 	$(call clean_ipkg,$(1))
 	$(call mklibs,$(1))
 	$(if $(SOURCE_DATE_EPOCH),find $(1)/ -mindepth 1 -execdir touch -hcd "@$(SOURCE_DATE_EPOCH)" "{}" +)
-- 
2.43.5

