From 41292e8bd135e0f1d293deb2cde4973ca0e763e5 Mon Sep 17 00:00:00 2001
From: sbwml <admin@cooluc.com>
Date: Thu, 26 Sep 2024 04:48:45 +0800
Subject: [PATCH] tools: add llvm/clang toolchain

Signed-off-by: sbwml <admin@cooluc.com>
---
 tools/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tools/Makefile b/tools/Makefile
index 40c3ec1..c7f970b 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -29,6 +29,9 @@ endif
 ifneq ($(CONFIG_SDK)$(CONFIG_TARGET_INITRAMFS_COMPRESSION_LZO),)
   BUILD_LZO_TOOLS = y
 endif
+ifeq ($(or $(filter clang,$(CONFIG_KERNEL_CC)),$(filter y,$(CONFIG_BPF_TOOLCHAIN_HOST))),)
+  BUILD_CLANG_HOST = y
+endif
 
 tools-y += autoconf
 tools-y += autoconf-archive
@@ -83,6 +86,7 @@ tools-$(if $(CONFIG_BUILD_ALL_HOST_TOOLS)$(CONFIG_TARGET_tegra),y) += cbootimage
 tools-$(if $(CONFIG_BUILD_ALL_HOST_TOOLS)$(CONFIG_USES_MINOR),y) += kernel2minor
 tools-$(if $(CONFIG_BUILD_ALL_HOST_TOOLS)$(CONFIG_USE_SPARSE),y) += sparse
 tools-$(if $(CONFIG_BUILD_ALL_HOST_TOOLS)$(CONFIG_USE_LLVM_BUILD),y) += llvm-bpf
+tools-$(if $(BUILD_CLANG_HOST),y) += clang
 
 # builddir dependencies
 $(curdir)/autoconf/compile := $(curdir)/m4/compile
-- 
2.43.5

