diff --git a/toolchain/gcc/Config.in b/toolchain/gcc/Config.in
index ca89051..510c813 100644
--- a/toolchain/gcc/Config.in
+++ b/toolchain/gcc/Config.in
@@ -17,6 +17,9 @@ choice
 
 	config GCC_USE_VERSION_14
 		bool "gcc 14.x"
+
+	config GCC_USE_VERSION_15
+		bool "gcc 15.x"
 endchoice
 
 config GCC_USE_GRAPHITE
diff --git a/toolchain/gcc/Config.version b/toolchain/gcc/Config.version
index 200a356..5675b9a 100644
--- a/toolchain/gcc/Config.version
+++ b/toolchain/gcc/Config.version
@@ -10,11 +10,16 @@ config GCC_VERSION_14
 	default y if GCC_USE_VERSION_14
 	bool
 
+config GCC_VERSION_15
+	default y if GCC_USE_VERSION_15
+	bool
+
 config GCC_VERSION
 	string
 	default "11.3.0"	if GCC_VERSION_11
 	default "13.2.0"	if GCC_VERSION_13
 	default "14.1.0"	if GCC_VERSION_14
+	default "15.0.0"	if GCC_VERSION_15
 	default "12.3.0"
 
 config GCC_USE_DEFAULT_VERSION
diff --git a/toolchain/gcc/common.mk b/toolchain/gcc/common.mk
index f5db99f..24c6535 100644
--- a/toolchain/gcc/common.mk
+++ b/toolchain/gcc/common.mk
@@ -26,7 +26,11 @@ PKG_VERSION:=$(firstword $(subst +, ,$(GCC_VERSION)))
 GCC_MAJOR_VERSION:=$(word 1,$(subst ., ,$(PKG_VERSION)))
 GCC_DIR:=$(PKG_NAME)-$(PKG_VERSION)
 
-PKG_SOURCE_URL:=@GNU/gcc/gcc-$(PKG_VERSION)
+ifeq ($(PKG_VERSION),15.0.0)
+  PKG_SOURCE_URL:=https://github.com/sbwml/gcc/releases/download/gcc-$(PKG_VERSION)
+else
+  PKG_SOURCE_URL:=@GNU/gcc/gcc-$(PKG_VERSION)
+endif
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
 PKG_CPE_ID:=cpe:/a:gnu:gcc
 
@@ -46,6 +50,10 @@ ifeq ($(PKG_VERSION),14.1.0)
   PKG_HASH:=e283c654987afe3de9d8080bc0bd79534b5ca0d681a73a11ff2b5d3767426840
 endif
 
+ifeq ($(PKG_VERSION),15.0.0)
+  PKG_HASH:=c91bf47b6caf585085e7f44b7b14ce4f10cbc9ce10be90098c7ebb0dabb66b3a
+endif
+
 PATCH_DIR=../patches-$(GCC_MAJOR_VERSION).x
 
 BUGURL=http://bugs.openwrt.org/
