diff --git a/package/firmware/linux-firmware/Makefile b/package/firmware/linux-firmware/Makefile
index f0d3e85..29c4808 100644
--- a/package/firmware/linux-firmware/Makefile
+++ b/package/firmware/linux-firmware/Makefile
@@ -8,12 +8,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=linux-firmware
-PKG_VERSION:=20220411
+PKG_VERSION:=20221214
 PKG_RELEASE:=1
 
 PKG_SOURCE_URL:=@KERNEL/linux/kernel/firmware
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
-PKG_HASH:=020b11f6412f4956f5a6f98de7d41867d2b30ea0ce81b1e2d206ec9840363849
+PKG_HASH:=e793783e92acbde549965521462d1d1327827360664cf242dbda08f075654331
 
 PKG_MAINTAINER:=Felix Fietkau <nbd@nbd.name>
 
