From b557005f9d3164de1f23a0239647f25d07b0b389 Mon Sep 17 00:00:00 2001
From: sbwml <admin@cooluc.com>
Date: Fri, 4 Oct 2024 10:47:41 +0800
Subject: [PATCH 03/10] rootfs: add r/w permissions for UCI configuration files

Signed-off-by: sbwml <admin@cooluc.com>
---
 include/rootfs.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/rootfs.mk b/include/rootfs.mk
index 574e3fe..00203a4 100644
--- a/include/rootfs.mk
+++ b/include/rootfs.mk
@@ -125,6 +125,7 @@ define prepare_rootfs
 			$(STAGING_DIR_HOST)/bin/upx --lzma --best "$(1)$$file" || true; \
 		done < "$(TOPDIR)/upx_list.txt"; \
 	fi
+	chmod 600 $(1)/etc/config/*
 	$(call clean_ipkg,$(1))
 	$(call mklibs,$(1))
 	$(if $(SOURCE_DATE_EPOCH),find $(1)/ -mindepth 1 -execdir touch -hcd "@$(SOURCE_DATE_EPOCH)" "{}" +)
-- 
2.43.5

