From 6243d903b853abbc81a3258356155f207e2c0853 Mon Sep 17 00:00:00 2001
From: sbwml <admin@cooluc.com>
Date: Fri, 4 Oct 2024 10:58:29 +0800
Subject: [PATCH 07/10] include: kernel: add miss config for linux-6.11

Signed-off-by: sbwml <admin@cooluc.com>
---
 include/kernel-defaults.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/kernel-defaults.mk b/include/kernel-defaults.mk
index 6ec2e75..ea97998 100644
--- a/include/kernel-defaults.mk
+++ b/include/kernel-defaults.mk
@@ -108,6 +108,7 @@ define Kernel/SetNoInitramfs
 	echo 'CONFIG_INITRAMFS_SOURCE=""' >> $(LINUX_DIR)/.config.set
 	echo '# CONFIG_INITRAMFS_FORCE is not set' >> $(LINUX_DIR)/.config.set
 	echo "# CONFIG_INITRAMFS_PRESERVE_MTIME is not set" >> $(LINUX_DIR)/.config.set
+	echo '# CONFIG_FSCACHE is not set' >> $(LINUX_DIR)/.config.set
 # CLANG
 	mv $(LINUX_DIR)/.config.set $(LINUX_DIR)/.config.old
 	grep -v CONFIG_LTO $(LINUX_DIR)/.config.old > $(LINUX_DIR)/.config.set
-- 
2.43.5

