From 360c76c16bb63cd2a7a5eb188c063468ce99073c Mon Sep 17 00:00:00 2001
From: sbwml <admin@cooluc.com>
Date: Thu, 10 Oct 2024 21:40:05 +0800
Subject: [PATCH 09/10] kernel: add legacy cgroup v1 memory controller

Signed-off-by: sbwml <admin@cooluc.com>
---
 config/Config-kernel.in | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/config/Config-kernel.in b/config/Config-kernel.in
index 91678cf..eb0f000 100644
--- a/config/Config-kernel.in
+++ b/config/Config-kernel.in
@@ -953,6 +953,16 @@ if KERNEL_CGROUPS
 		  the kmem extension can use it to guarantee that no group of processes
 		  will ever exhaust kernel resources alone.
 
+	config KERNEL_MEMCG_V1
+		bool "Legacy cgroup v1 memory controller"
+		depends on KERNEL_MEMCG
+		help
+		Legacy cgroup v1 memory controller which has been deprecated by
+		cgroup v2 implementation. The v1 is there for legacy applications
+		which haven't migrated to the new cgroup v2 interface yet. If you
+		do not have any such application then you are completely fine leaving
+		this option disabled.
+
 	config KERNEL_CGROUP_PERF
 		bool "Enable perf_event per-cpu per-container group (cgroup) monitoring"
 		select KERNEL_PERF_EVENTS
-- 
2.43.5

