From bd8a486bf2a805a7783b3a3e329b5756c1ff6127 Mon Sep 17 00:00:00 2001
From: sbwml <admin@cooluc.com>
Date: Sat, 12 Oct 2024 08:36:46 +0800
Subject: [PATCH] kernel: add PREEMPT_RT support for aarch64/x86_64

Signed-off-by: sbwml <admin@cooluc.com>
---
 config/Config-kernel.in | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/config/Config-kernel.in b/config/Config-kernel.in
index 39a9730..5439219 100644
--- a/config/Config-kernel.in
+++ b/config/Config-kernel.in
@@ -20,6 +20,19 @@ config KERNEL_BUILD_DOMAIN
 	  returned by 'uname -a' on running systems.
 	  If not set, uses system hostname at build time.
 
+config KERNEL_PREEMPT_RT
+	bool "Fully Preemptible Kernel (Real-Time)"
+	depends on (aarch64 || x86_64) && LINUX_6_12
+	help
+	  This option turns the kernel into a real-time kernel by replacing
+	  various locking primitives (spinlocks, rwlocks, etc.) with
+	  preemptible priority-inheritance aware variants, enforcing
+	  interrupt threading and introducing mechanisms to break up long
+	  non-preemptible sections. This makes the kernel, except for very
+	  low level and critical code paths (entry code, scheduler, low
+	  level interrupt handling) fully preemptible and brings most
+	  execution contexts under scheduler control.
+
 config KERNEL_PRINTK
 	bool "Enable support for printk"
 	default y
-- 
2.43.5

