diff --git a/src/layer_cake.qos.help b/src/layer_cake.qos.help
index 1cab3ed..c75bf8a 100644
--- a/src/layer_cake.qos.help
+++ b/src/layer_cake.qos.help
@@ -1,4 +1,4 @@
-This uses the cake qdisc as a replacement for both htb as shaper and fq_codel as leaf qdisc.
-This exercises cake's diffserv profile(s) as different "layers" of priority.
-This script requires that cake is selected as qdisc, and forces its usage.
-See: http://www.bufferbloat.net/projects/codel/wiki/Cake for more information
+这个 cake 列队规则使用 HTB 作为过滤器，使用 fq_codel 作为叶列队规则。
+这个 cake 规则将不同的文件分为不同的“层次”优先级。
+该脚本需要将该 cake 选为列队规则。
+请参阅：http://www.bufferbloat.net/projects/codel/wiki/Cake 获取更多信息
diff --git a/src/piece_of_cake.qos.help b/src/piece_of_cake.qos.help
index b95e9be..8bb0d31 100644
--- a/src/piece_of_cake.qos.help
+++ b/src/piece_of_cake.qos.help
@@ -1,4 +1,4 @@
-This just uses the cake qdisc as a replacement for both htb as shaper and fq_codel as leaf qdisc.
-It just does not come any simpler than this, in other words it truely is a "piece of cake".
-This script requires that cake is selected as qdisc, and forces its usage.
-See: http://www.bufferbloat.net/projects/codel/wiki/Cake for more information
+这个 cake 列队规则使用 HTB 作为过滤器，使用 fq_codel 作为叶列队规则。
+它不会比这更简单，换句话说，它真的是“小菜一碟”。
+该脚本需要将 cake 选为列队规则。
+请参阅：http://www.bufferbloat.net/projects/codel/wiki/Cake 获取更多信息
diff --git a/src/simple.qos.help b/src/simple.qos.help
index b3c0096..94a146a 100644
--- a/src/simple.qos.help
+++ b/src/simple.qos.help
@@ -1 +1 @@
-BW-limited three-tier prioritisation scheme with your qdisc on each queue. (default)
+使用 fq_codel 列队规则在每个列队上进行三层优先级的带宽控制。（默认）
diff --git a/src/simplest.qos.help b/src/simplest.qos.help
index c359256..8b37d9f 100644
--- a/src/simplest.qos.help
+++ b/src/simplest.qos.help
@@ -1 +1 @@
-Simplest possible configuration: HTB rate limiter with your qdisc attached.
+最简单的配置：使用带有 HTB 过滤器的列队规则来进行速率限制。
diff --git a/src/simplest_tbf.qos.help b/src/simplest_tbf.qos.help
index 3f93f89..f4936e2 100644
--- a/src/simplest_tbf.qos.help
+++ b/src/simplest_tbf.qos.help
@@ -1,2 +1 @@
-Simplest possible configuration (TBF): TBF rate limiter with your qdisc attached.
-TBF may give better performance than HTB on some architectures.
+最简单的配置（TBF）：使用带有 TBF 过滤器的列队规则来进行速率限制。在某些架构上，TBF 可能会比 HTB 提供更好的性能。
