diff --git a/package/kernel/gpio-button-hotplug/src/gpio-button-hotplug.c b/package/kernel/gpio-button-hotplug/src/gpio-button-hotplug.c
index 9876dee..87828e0 100644
--- a/package/kernel/gpio-button-hotplug/src/gpio-button-hotplug.c
+++ b/package/kernel/gpio-button-hotplug/src/gpio-button-hotplug.c
@@ -522,7 +522,7 @@ static int gpio_keys_button_probe(struct platform_device *pdev,
 				/* or the legacy (button->gpio is good) way? */
 				error = devm_gpio_request_one(dev,
 					button->gpio, GPIOF_IN | (
-					button->active_low ? GPIOF_ACTIVE_LOW :
+					button->active_low ? GPIOD_OUT_LOW :
 					0), desc);
 				if (error) {
 					dev_err_probe(dev, error,
@@ -674,7 +674,11 @@ static void gpio_keys_irq_close(struct gpio_keys_button_dev *bdev)
 	}
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 11, 0)
+static void gpio_keys_remove(struct platform_device *pdev)
+#else
 static int gpio_keys_remove(struct platform_device *pdev)
+#endif
 {
 	struct gpio_keys_button_dev *bdev = platform_get_drvdata(pdev);
 
@@ -685,7 +689,9 @@ static int gpio_keys_remove(struct platform_device *pdev)
 	else
 		gpio_keys_irq_close(bdev);
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0)
 	return 0;
+#endif
 }
 
 static struct platform_driver gpio_keys_driver = {
