diff --git a/include/target.mk b/include/target.mk
index b5e3e7f..d655af9 100644
--- a/include/target.mk
+++ b/include/target.mk
@@ -264,6 +264,10 @@ ifeq ($(DUMP),1)
     CPU_TYPE ?= riscv64
     CPU_CFLAGS_riscv64:=-mabi=lp64d -march=rv64imafdc
   endif
+  ifeq ($(BOARD),rockchip)
+    CPU_CFLAGS = -O2 -Wl,--gc-sections -pipe
+    CPU_CFLAGS_generic = -march=armv8-a+crc+crypto
+  endif
   ifneq ($(CPU_TYPE),)
     ifndef CPU_CFLAGS_$(CPU_TYPE)
       $(warning CPU_TYPE "$(CPU_TYPE)" doesn't correspond to a known type)
