From 01e9c5d991cbe0dad0283e102be9c5570f4549f6 Mon Sep 17 00:00:00 2001
From: sbwml <admin@cooluc.com>
Date: Tue, 22 Oct 2024 21:00:28 +0800
Subject: [PATCH] dnsmasq: drop 'extraconftext' parameter

Signed-off-by: sbwml <admin@cooluc.com>
---
 .../services/dnsmasq/files/dnsmasq.init       | 23 ++++++-------------
 1 file changed, 7 insertions(+), 16 deletions(-)

diff --git a/package/network/services/dnsmasq/files/dnsmasq.init b/package/network/services/dnsmasq/files/dnsmasq.init
index b864ea9069..96110c2b38 100755
--- a/package/network/services/dnsmasq/files/dnsmasq.init
+++ b/package/network/services/dnsmasq/files/dnsmasq.init
@@ -12,7 +12,6 @@ ADD_WAN_FQDN=0
 ADD_LOCAL_FQDN=""
 
 BASECONFIGFILE="/var/etc/dnsmasq.conf"
-EXTRACONFFILE="extraconfig.conf"
 BASEHOSTFILE="/tmp/hosts/dhcp"
 TRUSTANCHORSFILE="/usr/share/dnsmasq/trust-anchors.conf"
 TIMEVALIDFILE="/var/state/dnsmasqsec"
@@ -1145,21 +1144,13 @@ dnsmasq_start()
 	xappend "--dhcp-broadcast=tag:needs-broadcast"
 
 
-	# Create a dnsmasq.d dir for each instance
-	config_get dnsmasqconfdir "$cfg" confdir "/tmp/dnsmasq${cfg:+.$cfg}.d"
-	# Ensure dnsmasqconfdir is an absolute path
-	[ "${dnsmasqconfdir:0:1}" = '/' ] && {
-		xappend "--conf-dir=$dnsmasqconfdir"
-		dnsmasqconfdir="${dnsmasqconfdir%%,*}"
-		[ ! -d "$dnsmasqconfdir" ] && mkdir -p "$dnsmasqconfdir"
-		xappend "--user=dnsmasq"
-		xappend "--group=dnsmasq"
-		echo >> "$CONFIGFILE_TMP"
-
-		# EXTRACONFFILE allows new dnsmasq parameters before they are natively handled in this init file
-		config_get extraconftext "$cfg" extraconftext
-		[ -n "$extraconftext" ] && echo -e "$extraconftext" > "$dnsmasqconfdir"/"$EXTRACONFFILE"
-	}
+	config_get dnsmasqconfdir "$cfg" confdir "/tmp/dnsmasq.d"
+	xappend "--conf-dir=$dnsmasqconfdir"
+	dnsmasqconfdir="${dnsmasqconfdir%%,*}"
+	[ ! -d "$dnsmasqconfdir" ] && mkdir -p "$dnsmasqconfdir"
+	xappend "--user=dnsmasq"
+	xappend "--group=dnsmasq"
+	echo >> "$CONFIGFILE_TMP"
 
 	config_get_bool enable_tftp "$cfg" enable_tftp 0
 	[ "$enable_tftp" -gt 0 ] && {
-- 
2.42.0

