--- a/drivers/net/ovpn-dco/main.c
+++ b/drivers/net/ovpn-dco/main.c
@@ -125,7 +125,11 @@ static void ovpn_setup(struct net_device
 	const int overhead = sizeof(u32) + NONCE_WIRE_SIZE + 16 + sizeof(struct udphdr) +
 			     max(sizeof(struct ipv6hdr), sizeof(struct iphdr));
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0)
+	netdev_features_t feat = NETIF_F_SG | dev->lltx |
+#else
 	netdev_features_t feat = NETIF_F_SG | NETIF_F_LLTX |
+#endif
 				 NETIF_F_HW_CSUM | NETIF_F_RXCSUM | NETIF_F_GSO |
 				 NETIF_F_GSO_SOFTWARE | NETIF_F_HIGHDMA;
 
