--- a/kernel/pf_ring.c
+++ b/kernel/pf_ring.c
@@ -5196,7 +5196,11 @@ static int is_netdev_promisc(struct net_
   debug_printk(1, "checking promisc for %s\n", netdev->name);
 
   rtnl_lock();
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,17,0)
   if_flags = (short) dev_get_flags(netdev);
+#else
+  if_flags = (short) netif_get_flags(netdev);
+#endif
   rtnl_unlock();
 
   return !!(if_flags & IFF_PROMISC);
@@ -5211,7 +5215,11 @@ static void set_netdev_promisc(struct ne
 
   rtnl_lock();
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,17,0)
   if_flags = (short) dev_get_flags(netdev);
+#else
+  if_flags = (short) netif_get_flags(netdev);
+#endif
   if(!(if_flags & IFF_PROMISC)) {
     if_flags |= IFF_PROMISC;
 #if(LINUX_VERSION_CODE < KERNEL_VERSION(5,0,0) && \
@@ -5234,7 +5242,11 @@ static void unset_netdev_promisc(struct
 
   rtnl_lock();
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,17,0)
   if_flags = (short) dev_get_flags(netdev);
+#else
+  if_flags = (short) netif_get_flags(netdev);
+#endif
   if(if_flags & IFF_PROMISC) {
     if_flags &= ~IFF_PROMISC;
 #if(LINUX_VERSION_CODE < KERNEL_VERSION(5,0,0) && \
