--- a/extensions/LUA/prot_buf_ip.c
+++ b/extensions/LUA/prot_buf_ip.c
@@ -16,8 +16,14 @@
  *	along with this program; if not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <linux/version.h>
+
 #if defined(__KERNEL__)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,6,0)
+	#include <linux/bitops.h>
+#else
 	#include <net/checksum.h>
+#endif
 	#include <net/tcp.h>
 #endif
 
--- a/extensions/LUA/prot_buf_tcp.c
+++ b/extensions/LUA/prot_buf_tcp.c
@@ -16,8 +16,14 @@
  *	along with this program; if not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <linux/version.h>
+
 #if defined(__KERNEL__)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,6,0)
+	#include <linux/bitops.h>
+#else
 	#include <net/checksum.h>
+#endif
 	#include <net/tcp.h>
 #endif
 #include "controller.h"
--- a/extensions/LUA/prot_buf_udp.c
+++ b/extensions/LUA/prot_buf_udp.c
@@ -16,9 +16,15 @@
  *	along with this program; if not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <linux/version.h>
+
 #if defined(__KERNEL__)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,6,0)
+	#include <linux/bitops.h>
+#else
 	#include <net/checksum.h>
 #endif
+#endif
 
 #include "controller.h"
 
