--- a/nat46/modules/nat46-netdev.c
+++ b/nat46/modules/nat46-netdev.c
@@ -110,7 +110,11 @@ static void nat46_netdev_setup(struct ne
 	dev->hard_header_len = 0;
 	dev->addr_len = 0;
 	dev->mtu = 16384; /* iptables does reassembly. Rather than using ETH_DATA_LEN, let's try to get as much mileage as we can with the Linux stack */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 12, 0)
 	dev->features = NETIF_F_NETNS_LOCAL;
+#else
+	dev->features = dev->netns_local;
+#endif
 	dev->flags = IFF_NOARP | IFF_POINTOPOINT;
 }
 
