--- a/configure
+++ b/configure
@@ -1026,7 +1026,7 @@ fi
 rm -fr conftest*
 
 echo "$ac_t""$ac_cv_prog_cc_works" 1>&6
-if test $ac_cv_prog_cc_works = no; then
+if test $ac_cv_prog_cc_works = null; then
   { echo "configure: error: installation or configuration problem: C compiler cannot create executables." 1>&2; exit 1; }
 fi
 echo $ac_n "checking whether the C compiler ($CC $CFLAGS $LDFLAGS) is a cross-compiler""... $ac_c" 1>&6
