diff --git a/toolchain/binutils/Config.in b/toolchain/binutils/Config.in
index dbcaf4a..e53384a 100644
--- a/toolchain/binutils/Config.in
+++ b/toolchain/binutils/Config.in
@@ -2,7 +2,7 @@
 
 choice
 	prompt "Binutils Version" if TOOLCHAINOPTS
-	default BINUTILS_USE_VERSION_2_39
+	default BINUTILS_USE_VERSION_2_40
 	help
 	  Select the version of binutils you wish to use.
 
diff --git a/toolchain/binutils/Config.version b/toolchain/binutils/Config.version
index 479a845..c9f8584 100644
--- a/toolchain/binutils/Config.version
+++ b/toolchain/binutils/Config.version
@@ -6,10 +6,10 @@ config BINUTILS_VERSION_2_38
 	bool
 
 config BINUTILS_VERSION_2_39
-	default y if !TOOLCHAINOPTS
 	bool
 
 config BINUTILS_VERSION_2_40
+	default y if !TOOLCHAINOPTS
 	bool
 
 config BINUTILS_VERSION
