From 8ff2c79869c1ab3aa3f9c565197111809a482599 Mon Sep 17 00:00:00 2001
From: Todd Short <tshort@akamai.com>
Date: Thu, 29 Aug 2019 20:21:58 -0400
Subject: [PATCH 10/43] QUIC: Quick fix: s2c to c2s for early secret

---
 ssl/ssl_quic.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/ssl/ssl_quic.c
+++ b/ssl/ssl_quic.c
@@ -207,7 +207,7 @@ int quic_set_encryption_secrets(SSL *ssl
     /* secrets from the POV of the client */
     switch (level) {
     case ssl_encryption_early_data:
-        s2c_secret = ssl->early_secret;
+        c2s_secret = ssl->early_secret;
         break;
     case ssl_encryption_handshake:
         c2s_secret = ssl->client_hand_traffic_secret;
