diff --git a/drivers/char/lrng/lrng_es_mgr.c b/drivers/char/lrng/lrng_es_mgr.c
index b6a63172b..e577035ff 100644
--- a/drivers/char/lrng/lrng_es_mgr.c
+++ b/drivers/char/lrng/lrng_es_mgr.c
@@ -393,7 +393,7 @@ void __init lrng_rand_initialize_early(void)
 	unsigned int i;
 
 	for (i = 0; i < ARRAY_SIZE(seed.data); i += longs) {
-		longs = arch_get_random_seed_longs_early(seed.data + i,
+		longs = arch_get_random_seed_longs(seed.data + i,
 						   ARRAY_SIZE(seed.data) - i);
 		if (longs)
 			continue;
