diff --git a/package/libs/libnl-tiny/Makefile b/package/libs/libnl-tiny/Makefile
index 317d24b..2739929 100644
--- a/package/libs/libnl-tiny/Makefile
+++ b/package/libs/libnl-tiny/Makefile
@@ -12,9 +12,9 @@ PKG_RELEASE:=1
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL=$(PROJECT_GIT)/project/libnl-tiny.git
-PKG_SOURCE_DATE:=2021-11-21
-PKG_SOURCE_VERSION:=8e0555fb39f51a5d6436b4f1370850caa03611ea
-PKG_MIRROR_HASH:=2cfbcc62384535674a2c0157cb24a0736520fcb66ed50be23bf9141c8488885f
+PKG_SOURCE_DATE:=2022-11-01
+PKG_SOURCE_VERSION:=db3b2cdbca5277723326a2720024a59fb821ae36
+PKG_MIRROR_HASH:=f6f9e2cb5366cda3be6d27a6320aa6cf3681c0b10df07b29ab4e4005e9f04f1c
 CMAKE_INSTALL:=1
 
 PKG_LICENSE:=LGPL-2.1
