diff --git a/net/miniupnpd/files/miniupnpd.init b/net/miniupnpd/files/miniupnpd.init
index de3504529..e89e4e55a 100644
--- a/net/miniupnpd/files/miniupnpd.init
+++ b/net/miniupnpd/files/miniupnpd.init
@@ -217,6 +217,8 @@ stop_service() {
 }
 
 start_service() {
+	uci set upnpd.config.presentation_url="http://$(uci -q get network.lan.ipaddr)/"
+	uci commit upnpd
 	config_load "upnpd"
 	config_foreach upnpd "upnpd"
 }
