diff --git a/package/kernel/linux/modules/fs.mk b/package/kernel/linux/modules/fs.mk
index 91cf1ea..14aa139 100644
--- a/package/kernel/linux/modules/fs.mk
+++ b/package/kernel/linux/modules/fs.mk
@@ -498,6 +498,27 @@ endef
 $(eval $(call KernelPackage,fs-ntfs))
 
 
+define KernelPackage/fs-ntfs3
+  SUBMENU:=$(FS_MENU)
+  TITLE:=NTFS3 Read-Write file system support
+  DEPENDS:=+kmod-nls-base
+  KCONFIG:= \
+	CONFIG_NTFS3_FS \
+	CONFIG_NTFS3_64BIT_CLUSTER=y \
+	CONFIG_NTFS3_LZX_XPRESS=y \
+	CONFIG_NTFS3_FS_POSIX_ACL=y
+  FILES:=$(LINUX_DIR)/fs/ntfs3/ntfs3.ko
+  AUTOLOAD:=$(call AutoLoad,30,ntfs3)
+  $(call AddDepends/nls)
+endef
+
+define KernelPackage/fs-ntfs3/description
+ Kernel module for NTFS3 filesystem support
+endef
+
+$(eval $(call KernelPackage,fs-ntfs3))
+
+
 define KernelPackage/pstore
   SUBMENU:=$(FS_MENU)
   TITLE:=Pstore file system
