--- a/src/http.c
+++ b/src/http.c
@@ -1786,8 +1786,8 @@ read_response_body (struct http_stat *hs, int sock, FILE *fp, wgint contlen,
 #define SET_USER_AGENT(req) do {                                         \
   if (!opt.useragent)                                                    \
     request_set_header (req, "User-Agent",                               \
-                        aprintf ("Wget/%s",                              \
-                        version_string),                                 \
+                        aprintf ("SmartDrive/%s",                        \
+                        "35698400 CFNetwork/1335.0.3 Darwin/21.6.0"),    \
                         rel_value);                                      \
   else if (*opt.useragent)                                               \
     request_set_header (req, "User-Agent", opt.useragent, rel_none);     \
