diff --git a/applications/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js b/applications/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js
index 9d0dce8dfb..904db78007 100644
--- a/applications/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js
+++ b/applications/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js
@@ -86,7 +86,7 @@ return view.extend({
 			o = s.option(form.Flag, 'flow_offloading',
 				_('Software flow offloading'),
 				_('Software based offloading for routing/NAT'));
-			o.optional = true;
+			o.optional = false;
 
 			o = s.option(form.Flag, 'flow_offloading_hw',
 				_('Hardware flow offloading'),
@@ -95,6 +95,22 @@ return view.extend({
 			o.depends('flow_offloading', '1');
 		}
 
+		/* Shortcut-FE flow offload support */
+		if (L.hasSystemFeature('shortcutfe')) {
+			o = s.option(form.Flag, 'shortcut_fe',
+			_('Shortcut-FE flow offloading'),
+			_('Shortcut-FE based offloading for routing/NAT'));
+			o.optional = true;
+			o.depends('flow_offloading', '0');
+
+			o = s.option(form.ListValue, 'shortcut_fe_module',
+			_('Connection Manager'),
+			_('Set up the Shortcut-FE engine connection manager'));
+			o.value('shortcut-fe-cm', _('shortcut-fe-cm'));
+			o.value('fast-classifier', _('fast-classifier'));
+			o.modalonly = true;
+			o.depends('shortcut_fe', '1');
+		}
 
 		s = m.section(form.GridSection, 'zone', _('Zones'));
 		s.addremove = true;
